wgpu-native/src/instance.rs

```diff
@@ -301,11 +302,16 @@ pub extern "C" fn wgpu_create_surface_from_windows_hwnd(
     ))
 }
 
-pub fn request_adapter(
+pub type RequestAdapterCallback =
+    extern "C" fn(adapter: *const AdapterId, userdata: *mut c_void);
```

Comment:
I noticed we use `*mut u8` for `userdata` in buffer mapping, but maybe we should consider using `*mut c_void` like we do here. For example, we expect `userdata` to generally be a user-provided type anyway (not necessarily a byte array), so is there any benefit from requiring `*mut u8`?

Reply:
yes, I agree, *mut void seems more appropriate. We might need to file a PR to upstream webgpu header